wallet/wallet-controller/src/lib.rs

```diff
@@ -618,6 +647,31 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static, W: WalletEvents> Controll
             }
 
             tokio::time::sleep(NORMAL_DELAY).await;
+
+            self.rebroadcast_txs(&mut rebroadcast_txs_timer);
+        }
+    }
+
+    /// Rebroadcast not confirmed transactions
+    fn rebroadcast_txs(&mut self, rebroadcast_txs_timer: &mut Instant) {
+        if Instant::now() >= *rebroadcast_txs_timer {
+            let res = self.wallet.get_transactions_to_be_broadcasted().map(|txs| async {
+                for tx in txs {
+                    let tx_id = tx.transaction().get_id();
+                    let res = self.rpc_client.submit_transaction(tx).await;
+                    if let Err(e) = res {
+                        log::error!("Rebroadcasting for tx {tx_id} failed: {e}");
+                    }
```

Comment:
Is `log::error!` maybe a bit too harsh? I can imagine some errors are serious, like malformed transaction, but others are fine as long as we retry broadcasting later (e.g. insufficient fees or something like that)

Reply:
changed

Reply:
changed to warn